idelib/dataset.py

```diff
@@ -735,8 +737,32 @@ def children(self):
 
 
     def __repr__(self):
-        return '<%s %d: %r at 0x%08x>' % (self.__class__.__name__, 
-                                          self.id, self.path(), id(self))
+        try:
+            if self._unitsStr is None:
+                # If there's no cached string of measurement types and units, make it.
+                if not self.children:
+                    units = [self.units]
+                else:
+                    units = [c.units for c in self.children]
+
+                    # If all subchannel units are the same, show only one.
+                    if len(set(units)) == 1:  # Cheesy test for identical units tuples
+                        units = [units[0]]
+
+                names = []
+                for measurement, name in units:
+                    if name:
+                        name = "(%s)" % name
+                    names.append(("%s %s" % (measurement, name)).strip())
+                self._unitsStr = ", ".join(names)
+
+            s = (": %s" % self._unitsStr) if self._unitsStr else ""
+            return '<%s %d %r%s>' % (self.__class__.__name__,
+                                     self.id, self.path(), s)
+
+        except (AttributeError, TypeError, ValueError):
+            # Just in case, so __repr__() never completely fails.
+            return object.__repr__(self)
```

Comment:
It shouldn't ever fail, so this clause should never run; it's just a CYA move in case something unexpected happens. I don't think it's worth a lot of work.

Reply:
that's fair, I just figured it'd be worth copy-pasting the old one/two lines to the catch statement

Reply:
That's a good point I hadn't considered, makes sense.

Reply:
seems like this was resolved? please reopen if I'm mistaken